Use cartId from REST URL

elzekool opened the pull request:
Resolution for Guest cart API ignoring cartId in url for some methods

### Description
For /V1/carts/mine/items and /V1/carts/mine/items the quoteId from the
quote item is used. For guest cart this is confusing as it makes the
cartId in the URL useless. For logged in carts it makes it unsafe as it
allows adding products to any active cart.

### Fixed Issues
1. magento/magento2#14086: Guest cart API ignoring cartId in url for some methods

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

rogyar commented:
Hello, @elzekool. Thank you for your contribution. 
In order to preserve backward compatibility, adding new methods to the existing interfaces is prohibited. A new interface should be created instead.
Please, take a look at the http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/

elzekool committed:
[ #14086 ] Use cartId in REST URL

For /V1/carts/mine/items and /V1/carts/mine/items the quoteId from the
quote item is used. For guest cart this is confusing as it makes the
cartId in the URL useless. For logged in carts it makes it unsafe as it
allows adding products to any active cart.

Relates to: #14086

rogyar commented:
Hello @elzekool. Are you going to proceed working on this PR?

rogyar commented:
I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the contribution
